app/models/piece.rb

```diff
@@ -4,8 +4,82 @@ class Piece < ActiveRecord::Base
 
 	enum status: [:black, :white]
 
-  def is_obstructed(destination_x, destination_y)
+  def is_obstructed?(destination_x, destination_y)
+    x1, y1 = self.x, self.y
+    x2, y2 = destination_x, destination_y
```

Comment:
instead of using these variables. I'd suggest using `x`, `y`, `destination_x, & `destination_y`

Btw: `x` and `y` in this context are the same as `self.x` and `self.y` :smile: